wolfSSL-FIPS-FAQ/src/section02.md

```diff
@@ -0,0 +1,339 @@
+
+### Folloup POST Q: Who can determine when NO_ATTRIBUTE_CONSTRUCTOR is allowed?
+
+A: Only wolfSSL staff in collaboration with an NVLAP accredited FIPS lab during an OE operational testing effort can determine the viability of the feature NO_ATTRIBUTE_CONSTRUCTOR. If the feature is warranted for a specific operating environment:
+ 
+1) It will be noted in the FIPS user guide section for that operating environement
+
+2) The bundle releases the customer receives from wolfSSL will build cleanly with the feature enabled (no compile time error)
+
+3) The default settings for the module will turn the setting on.
+
+### Followup Post Q: What about with fips-ready, can I use NO_ATTRIBUTE_CONSTRUCTOR with fips-ready?
+
+A: The term "fips-ready' implies a module is abiding by ALL the FIPS requirements such that it could be submitted to the CMVP without any changes and achieve FIPS certification. Using the feature NO_ATTRIBUTE_CONSTRUCTOR without woldSSL staff and an NVLAP accredited FIPS lab approving it for that operating environement would imply the fips-ready solution is no longer a fips-ready release, that release would no longer be considered "fips-ready" until reviewed by wolfSSL staff in collaboration with an NVLAP accredited FIPS lab.
+
+## What can go wrong for the end user after basic testing?
+
+A: (Multi-part)
+
+a) The integrity test can fail due to corruption
+
+b) Power On Self TEst can fail
+	
+i) This is very unlikely after initial testing.
+
+ii) wolfSSL has seen cases where the customer receiving the binary had the wrong hardware for with PAA.
+	
+iii) wolfSSL has also seen a case where the hardware was correct but the with PAA feature (AESNI) was disabled in the BIOS settings of the end-point system from the factory. In this case, the end customer had to go into the BIOS and enable the AESNI feature on the chip for the FIPS module with PAA to not seg-fault on their system.
+	
+iv) Windows: ASLR (Address space layout randomization) is enabled on the computer where the test is failing meaning the OS is free to "chop up the FIPS module and scatter it about in memory" and somehow still expect the intgrity hash to be the same value. (This is impossible, please ensure ASLR is disabled on any PC working with the FIPS module or at a minimum ASLR is disabled for the apps/programs consuming the FIPS module actively)
+
+v) Windows: Randomized base addressing in Windows OS. Because the function addresses are included in the integrity check, the module needs to load at the same location in memory each time. There are 2 factors to this one: 
+
+1) Must diable Random base address at the project level. Example: https://github.com/wolfSSL/wolfssl/blob/master/IDE/WIN10/wolfssl-fips.vcxproj#L159
```

Comment:
Please use Markdown link syntax, so link shows up as a clickable link in generated output.